tests/test_mongodb_log_based_interruptible.py

```diff
@@ -0,0 +1,330 @@
+                             set(found_stream.get('metadata', {}).get('table-key-properties')))
+
+            # assert that the row counts are correct
+            self.assertEqual(self.expected_row_count_1()[found_stream['stream_name']],
+                             found_stream.get('metadata', {}).get('row-count'))
+
+
+        #  -----------------------------------
+        # -----------Initial Full Table Sync ---------
+        #  -----------------------------------
+        # Select simple_coll_1 and simple_coll_2 streams and add replication method metadata
+        for stream_catalog in found_catalogs:
+            annotated_schema = menagerie.get_annotated_schema(conn_id, stream_catalog['stream_id'])
+            additional_md = [{"breadcrumb": [], "metadata": {'replication-method': 'LOG_BASED'}}]
+            selected_metadata = connections.select_catalog_and_fields_via_metadata(conn_id,
+                                                                                   stream_catalog,
+                                                                                   annotated_schema,
+                                                                                   additional_md)
+
+        runner.run_sync_mode(self, conn_id)
+
+        # streams that we synced are the ones that we expect to see
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        records_by_stream = runner.get_records_from_target_output()
+
+        for tap_stream_id in self.expected_sync_streams():
+            self.assertGreaterEqual(record_count_by_stream[tap_stream_id], self.expected_row_count_1()[tap_stream_id])
+
+        initial_state = menagerie.get_state(conn_id)
+        bookmarks = initial_state['bookmarks']
+
+        self.assertIsNone(initial_state['currently_syncing'])
+
+        # verify each bookmark matches our expectation prior to setting the interrupted state
+        for table_name in self.expected_sync_streams():
+
+            table_bookmark = bookmarks['simple_db-'+table_name]
+            bookmark_keys = set(table_bookmark.keys())
+
+            self.assertIn('version', bookmark_keys)
+            self.assertIn('last_replication_method', bookmark_keys)
+            self.assertIn('initial_full_table_complete', bookmark_keys)
+
+            self.assertIn('oplog_ts_time', bookmark_keys)
+            self.assertIn('oplog_ts_inc', bookmark_keys)
+            self.assertNotIn('replication_key', bookmark_keys)
+
+            self.assertEqual('LOG_BASED', table_bookmark['last_replication_method'])
+            self.assertTrue(table_bookmark['initial_full_table_complete'])
+            self.assertIsInstance(table_bookmark['version'], int)
```

Comment:
I don't think these are necessary unless they are not covered by the standard oplog test.

Reply:
I wanted to verify if the log based replication has been set accurately in the interruptible test.